src/main/java/com/alipay/oceanbase/rpc/stream/ObTableClientQueryAsyncStreamResult.java

```diff
@@ -178,10 +224,31 @@ public boolean next() throws Exception {
             boolean hasNext = false;
             Iterator<Map.Entry<Long, ObPair<Long, ObTableParam>>> it = expectant.entrySet()
                 .iterator();
+            int retryTimes = 0;
             while (it.hasNext()) {
                 Map.Entry<Long, ObPair<Long, ObTableParam>> entry = it.next();
-                // try access new partition, async will not remove useless expectant
-                referToNewPartition(entry.getValue());
+                try {
+                    // try access new partition, async will not remove useless expectant
+                    referToNewPartition(entry.getValue());
+                } catch (Exception e) {
+                    if (e instanceof ObTableNeedFetchAllException) {
+                        this.asyncRequest.getObTableQueryRequest().getTableQuery()
+                            .adjustStartKey(currentStartKey);
```

Comment:
does adjustStartKey work in key-partitioned table?

Reply:
only range partitioned table is supported; other partitioning uses default behavior